### PR TITLE
Added Resize Settings and Resize Operations

### PR DESCRIPTION
**Pull request by worryg0d**
Done:
- added resize operations to the cluster data center status. It shows a customer the current resize operations for each data center.
- added resize settings to the cluster specs
- added validation of resize settings
- verbose some messages in the resource update handlers

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -98,7 +98,7 @@ run: manifests generate fmt vet ## Run a controller from your host.
 	go run ./main.go
 
 .PHONY: docker-build
-docker-build: manifests generate test ## Build docker image with the manager.
+docker-build: manifests generate #test ## Build docker image with the manager.
```
Comment (testisnullus) [inline Makefile]: Revert this change please

File: pkg/instaclustr/client.go [hunk 2]
```diff
@@ -352,6 +352,8 @@ func (c *Client) UpdateCassandra(id string, cassandra models.CassandraClusterAPI
 		return err
 	}
 
+	fmt.Printf("CASSANDRA SPEC UPDATE: %s\n", data)
+
```
Comment (testisnullus) [inline pkg/instaclustr/client.go:356]: Remove this code please

File: pkg/models/errors.go [hunk 3]
```diff
@@ -61,4 +61,5 @@ var (
 	ErrPrivateLinkSupportedOnlyForSingleDC      = errors.New("private link is only supported for a single data centre")
 	ErrPrivateLinkSupportedOnlyForAWS           = errors.New("private link is supported only for an AWS cloud provider")
 	ErrImmutableSpec                            = errors.New("resource specification is immutable")
+	ErrOnlySingleConcurrentResizeAvailable      = errors.New("only 1 concurrent resize is allowed")
```
Comment (OleksiienkoMykyta) [inline pkg/models/errors.go:64]: Maybe we should change message from "only 1 concurrent resize is allowed" to "only single concurrent resize is allowed"

**Review by OleksiienkoMykyta — APPROVED** [review]

**Review by testisnullus — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (worryg0d) [reply Makefile]: Fixed

File: pkg/instaclustr/client.go (hunk 2 above)
Comment (worryg0d) [reply pkg/instaclustr/client.go]: Removed

File: pkg/models/errors.go (hunk 3 above)
Comment (worryg0d) [reply pkg/models/errors.go]: Changed

**Review by tengu-alt — APPROVED** [review]

File: apis/clusters/v1beta1/cassandra_types.go [hunk 4]
```diff
@@ -181,9 +183,10 @@ func (cs *CassandraSpec) HasRestore() bool {
 	return false
 }
 
-func (cs *CassandraSpec) NewDCsUpdate() models.CassandraClusterAPIUpdate {
+func (cs *CassandraSpec) ToClusterUpdate() models.CassandraClusterAPIUpdate {
```
Comment (ribaraka) [inline apis/clusters/v1beta1/cassandra_types.go:186]:
```suggestion
func (cs *CassandraSpec) DCsUpdateToInstAPI() models.CassandraClusterAPIUpdate {
```
The Instaclustr APIv2 has multiple endpoints for updating various fields in a cluster specification. One of them is `ClusterSettings`. Therefore, I suggest specifically naming the update fields.

File: apis/clusters/v1beta1/structs.go [hunk 5]
```diff
@@ -204,6 +205,71 @@ type immutableDC struct {
...
+		iRS = append(iRS, &ResizeSettings{
+			NotifySupportContacts: rs.NotifySupportContacts,
+			Concurrency:           rs.Concurrency,
+		})
+	}
+
+	return iRS
+}
+
+type ReplaceOperationInfoV2 struct {
```
Comment (ribaraka) [inline apis/clusters/v1beta1/structs.go:260]: Just call it `ReplaceOperation`

File: pkg/instaclustr/client.go [hunk 6]
```diff
@@ -208,7 +208,7 @@ func (c *Client) GetRedisUser(id string) (*models.RedisUser, error) {
 	return userRedis, nil
 }
 
-func (c *Client) UpdateRedis(id string, r *models.RedisDataCentreUpdate) error {
+func (c *Client) UpdateRedis(id string, r *models.RedisUpdate) error {
```
Comment (ribaraka) [inline pkg/instaclustr/client.go:211]: The message with different update fields also applies to this part of the code

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: pkg/instaclustr/client.go (hunk 6 above)
Comment (worryg0d) [reply pkg/instaclustr/client.go]: Fixed

File: apis/clusters/v1beta1/structs.go (hunk 5 above)
Comment (worryg0d) [reply apis/clusters/v1beta1/structs.go]: Renamed

File: apis/clusters/v1beta1/cassandra_types.go (hunk 4 above)
Comment (worryg0d) [reply apis/clusters/v1beta1/cassandra_types.go:186]: Fixed

**Review by ribaraka — APPROVED** [review]

**Review by testisnullus — APPROVED** [review]
